server/controller/schedulesController.js

```diff
@@ -0,0 +1,121 @@
+const conn = require("../mariadb");
+const { StatusCodes } = require("http-status-codes");
+
+const getScheduleByPeriod = ({ start, end }) => [
+  "SELECT * FROM SmartDay.schedules WHERE date BETWEEN ? AND ?",
+  [start, end],
+];
+
+const getScheduleByMonth = (month) => [
+  "SELECT * FROM SmartDay.schedules WHERE DATE_FORMAT(date, '%Y-%m') = ?",
+  [month],
+];
+
+const getScheduleByDate = (date) => ["SELECT * FROM SmartDay.schedules WHERE date = ?", [date]];
+
+const getSchedulesQuery = ({ start, end, date, month }) => {
+  if (start && end) return getScheduleByPeriod({ start, end });
+  if (month) return getScheduleByMonth(month);
+  if (date) return getScheduleByDate(date);
+
+  return ["SELECT * FROM SmartDay.schedules", []];
+};
+
+const getAllSchedules = (req, res) => {
+  const { start, end, date, month } = req.query;
+  const [sql, values] = getSchedulesQuery({ start, end, date, month });
+
+  conn.query(sql, values, (err, results) => {
+    if (err) {
+      console.error(err);
+      return res.status(StatusCodes.INTERNAL_SERVER_ERROR).end();
+    }
+    return res.status(StatusCodes.OK).json(results);
+  });
+};
+
+const getScheduleById = (req, res) => {
+  const { id } = req.params;
+
+  let sql = `
+    SELECT title, date, start_time, end_time, detail, completed
+	  FROM schedules
+	  WHERE id = ?
+    `;
+  let values = [id];
```

Comment:
재할당이 일어나지 않는다면 const로 변수는 선언해주세요!
```suggestion
  const sql = `
    SELECT title, date, start_time, end_time, detail, completed
	  FROM schedules
	  WHERE id = ?
    `;
  const values = [id];
```